src/hooks.ts

```diff
@@ -18,3 +19,57 @@ export function useFlexSize() {
   const { size } = useContext(boxContext)
   return size
 }
+
+export function useNode() {
+  const { node } = useContext(boxContext)
+  return node
+}
+
+/**
+ * explicitly set the size of the box's yoga node
+ * @requires that the surrounding Flex-Element has `disableSizeRecalc` set to `true`
+ */
+export function useSetSize(): (width: number, height: number) => void {
+  const { requestReflow, scaleFactor } = useContext(flexContext)
+  const node = useNode()
+
+  const sync = useCallback(
+    (width: number, height: number) => {
+      if (node == null) {
+        throw new Error('yoga node is null. sync size is impossible')
+      }
+      node.setWidth(width * scaleFactor)
+      node.setHeight(height * scaleFactor)
+      requestReflow()
+    },
+    [node, requestReflow]
+  )
+
+  return sync
+}
+
+const helperVector = new Vector3()
+
+/**
+ * explicitly sync the yoga node size with a mesh's geometry and uniform global scale
+ * @requires that the surrounding Flex-Element has `disableSizeRecalc` set to `true`
+ */
+export function useSyncGeometrySize(): (mesh: Mesh) => void {
+  const setSize = useSetSize()
+  return useCallback(
+    (mesh: Mesh) => {
+      mesh.updateMatrixWorld()
+      helperVector.setFromMatrixScale(mesh.matrixWorld)
+
+      //since the scale is in global space but the box boundings are in local space, scaling can't be translated, thus a uniform scaling is required to have this work properly
+      if (Math.abs(helperVector.x - helperVector.y) > 0.001 || Math.abs(helperVector.y - helperVector.z) > 0.001) {
```

Comment:
This seems like a redundant logic of what's happening in `reflow()`:
https://github.com/pmndrs/react-three-flex/pull/49/files#diff-1ee9f750a449411472ab19801d11385063fbc2d45dd8dd8f1abd403259b681fcR277

Can this be refactored into a single function or removed and using a single one instead?
I like the fact that you can sync sizes without a React rerender, but `reflow()` should do that already.

Reply:
Yeah, reflow does that, but only if I didn't turn off the automatic size calculation. Which is intended in this case, since I maybe only want to change the size of one Box.

I will try to move the logic into one function 👍